add smart contract wallet support

ethernet8023 opened the pull request:
- replaces 'siwe' with viem/siwe
- viem siwe has no specific errors, so siwe validation error is replaced with a generic message

closes https://github.com/stella3d/atp-evm/issues/2

I'm not exactly sure how this works with multichain wallets where e.g. a signature is valid on one chain and not another - it's possible that people who want to verify someone else's link later would have to check by using a client (`getClient`) for the specific chainId that the `siweMessage` was created for.

Also, it might be prudent to add an ability to choose a chain to sign for - some smart contract wallets might have different "native chains", where signatures are cheaper for a first time signature on that "native chain", if they don't offer some sort of counterfactual wallet deployment where signatures can be validated without having to deploy the wallet to that specific chain.

That's just a nice-to-have, though, and this should let 1271 wallets like [Safe](https://safe.global/) or [Sequence](https://sequence.app/wallet), or 4337 wallets like the [coinbase smart wallet](https://www.base.org/builders/smart-wallet) link their DIDs!

ethernet8023 committed:
add smart contract wallet support
- replaces 'siwe' with viem/siwe
- viem siwe has no specific errors, so siwe validation error is replaced with a generic message

stella3d commented:
> I'm not exactly sure how this works with multichain wallets where e.g. a signature is valid on one chain and not another - it's possible that people who want to verify someone else's link later would have to check by using a client (`getClient`) for the specific chainId that the `siweMessage` was created for.

`chainId` is required by SIWE, so we record it, and according to the [SIWE spec](https://eips.ethereum.org/EIPS/eip-4361)
contract account `MUST` be resolved only on this chain.
<img width="437" alt="image" src="https://github.com/user-attachments/assets/3b237586-f8d1-4dc9-ad58-51655a8575d1" />

stella3d commented:
> Also, it might be prudent to add an ability to choose a chain to sign for - some smart contract wallets might have different "native chains", where signatures are cheaper for a first time signature on that "native chain", if they don't offer some sort of counterfactual wallet deployment where signatures can be validated without having to deploy the wallet to that specific chain.
> 
> That's just a nice-to-have, though, and this should let 1271 wallets like [Safe](https://safe.global/) or [Sequence](https://sequence.app/wallet), or 4337 wallets like the [coinbase smart wallet](https://www.base.org/builders/smart-wallet) link their DIDs!

right now, in my testing it will just use whatever chain's wallet is most recently used by the user - for instance i can choose to sign using a Base version of my wallet if that's what i connect. i figured i would defer chain selection stuff to the wallet, let me know if that seems off

stella3d commented on src/WalletConnector.tsx:
```diff
@@ -37,44 +37,51 @@ export const makeSiweStatement = (address: `0x${string}`, did: DefinedDidString)
     chainId,
     nonce: uid(24),
-  })
+    issuedAt: new Date()
```
does this produce the same serialization as before ?

arilotter replied:
Yes - a sample serialization before:
```
127.0.0.1:5173 wants you to sign in with your Ethereum account:
0x831dE831A64405aF965C67d6E0De2F9876fa2d99

Prove control of 0x831dE831A64405aF965C67d6E0De2F9876fa2d99 to link it to did:plc:awpmnhm4q4y62hwxukiwg6ry

URI: http://127.0.0.1:5173/
Version: 1
Chain ID: 1
Nonce: d8158f4c2ed1cb87c926609f
Issued At: 2025-06-29T21:58:14.004Z
```
and after
```
127.0.0.1:5173 wants you to sign in with your Ethereum account:
0x831dE831A64405aF965C67d6E0De2F9876fa2d99

Prove control of 0x831dE831A64405aF965C67d6E0De2F9876fa2d99 to link it to did:plc:awpmnhm4q4y62hwxukiwg6ry

URI: http://127.0.0.1:5173/
Version: 1
Chain ID: 1
Nonce: a7997c23c8ebaebde4a2d729
Issued At: 2025-06-29T21:57:20.534Z
```

here's a visual diff:
![image](https://github.com/user-attachments/assets/31599abc-b642-41ce-ab11-934d17941623)


Internally, the `siwe` package uses `        this.issuedAt = this.issuedAt || new Date().toISOString();`,
and this PR does `issuedAt: message.issuedAt.toISOString(),`
[`ab6b6ad` (#5)](https://github.com/stella3d/atp-evm/pull/5/commits/ab6b6adb78ce14ccf7a1ce28b0ae0a75eb650511#diff-47dca761fee94b61961fe0402e54c7f8aecf901259a1c6e669d93b9d4c9e89bcR58)

the only difference is where we call `toISOString` - message construction or signing time - but it's a pure function & will return the same result everywhere it's called.

arilotter commented:
> figured i would defer chain selection stuff to the wallet, let me know if that seems off

that makes sense. maybe I want to change getClient to instead asynchronously obtain a client for the chain we're verifying the message on.

ethernet8023 committed:
validate the nonce in the siwe message matches the nonce in the siwe struct

ethernet8023 committed:
always validate the siwe message on the chain it's meant for

before, we would validate against whatever chain the user currently had
their wallet connected to.

arilotter commented:
> > figured i would defer chain selection stuff to the wallet, let me know if that seems off
> 
> that makes sense. maybe I want to change getClient to instead asynchronously obtain a client for the chain we're verifying the message on.

done this, see [`ae2e114` (#5)](https://github.com/stella3d/atp-evm/pull/5/commits/ae2e1149f38e35efb761b3acc435c02b703e3603)

stella3d reviewed (approved):
thanks for the contribution! looks good
